operator/halo2ipa/halo2ipa.go

```diff
@@ -10,31 +10,35 @@ import "C"
 import "unsafe"
 
 // MaxProofSize 4KB
-const MaxProofSize =  8 * 1024;
+const MaxProofSize = 8 * 1024
 
 // MaxProofSize 4KB
-const MaxParamsSize =  8 * 1024;
+const MaxParamsSize = 8 * 1024
 
 // MaxConstraintSystemSize 2KB
-const MaxConstraintSystemSize = 2 * 1024;
+const MaxConstraintSystemSize = 2 * 1024
 
 // MaxVerificationKeySize 1KB
-const MaxVerifierKeySize = 1024;
+const MaxVerifierKeySize = 1024
 
 // MaxKzgParamsSize 4KB
-const MaxIpaParamsSize = 4 * 1024;
+const MaxIpaParamsSize = 4 * 1024
 
 // MaxPublicInputSize 4KB
-const MaxPublicInputSize = 4 * 1024;
+const MaxPublicInputSize = 4 * 1024
 
-//Merge all pointers into one array and send across interface
+// Merge all pointers into one array and send across interface
 func VerifyHalo2IpaProof(
-	proofBuffer [MaxProofSize]byte, proofLen_u32 uint32, 
-	csBuffer [MaxConstraintSystemSize]byte, csLen_u32 uint32, 
-	vkBuffer [MaxVerifierKeySize]byte, vkLen_u32 uint32, 
-	ipaParamBuffer [MaxIpaParamsSize]byte, ipaParamLen_u32 uint32, 
+	proofBuffer [MaxProofSize]byte, proofLen_u32 uint32,
+	csBuffer [MaxConstraintSystemSize]byte, csLen_u32 uint32,
+	vkBuffer [MaxVerifierKeySize]byte, vkLen_u32 uint32,
+	ipaParamBuffer [MaxIpaParamsSize]byte, ipaParamLen_u32 uint32,
```

Comment:
We should change the type of the parameters to byte[]. Also, maybe we could just have `verificationKeyBuffer` as a parameter instead of `csBuffer`, `vkBuffer` and `ipaParamBuffer` and then deserialize them directly in the rust code.